File: Problems/003-longest-substring-without-repeating-characters/longest-substring-without-repeating-characters.go
package problem003

func lengthOfLongestSubstring(s string) int {
	index := [128]int{}
	for i := range index {
		index[i] = -1
	}

	ans, left := 0, 0

	for i := 0; i < len(s); i++ {
		if index[s[i]] >= left {
			left = index[s[i]] + 1

		} else if i-left+1 > ans {
			ans = i - left + 1
		}
		index[s[i]] = i
	}
	return ans
}

File: Problems/003-longest-substring-without-repeating-characters/longest-substring-without-repeating-characters.py
class Solution:
    def lengthOfLongestSubstring(self, s: str) -> int:
        indices = [-1] * 127
        ans = 0
        length = 0
        start = 0
        for i, c in enumerate(s):
            length += 1
            j = ord(c)
            if indices[j] >= start:
                length = i - indices[j]
                start = indices[j] + 1
            indices[j] = i
            if length > ans:
                ans = length
        return ans